lib/makePromise.js

```diff
@@ -111,8 +111,12 @@ function extractPromisesFromObject(obj) {
     };
 });
 
-['resolve', 'reject'].forEach(function (staticMethodName) {
-    makePromise[staticMethodName] = Promise[staticMethodName];
+// Expose all of Bluebird's static methods, except the ones related to long stack traces,
+// unhandled rejections and the scheduler, which we need to manage ourselves:
+Object.keys(Promise).forEach(function (staticMethodName) {
+    if (!/^_|^on|^setScheduler|ongStackTraces/.test(staticMethodName) && typeof makePromise[staticMethodName] === 'undefined') {
```

Comment:
Done. It doesn't make a different currently (all properties are methods or error constructors), but I agree that it's a useful safeguard.